### PR TITLE
Update DG (class diagrams), food data and route parser

### LINKED ISSUE #163: [PE-D] Special letters/icons not shown
Could consider using the cmd output to show on the user guide and adding a note for what the "??" could represent.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: neojiaern/ped#3

### LINKED ISSUE #189: Change all class diagrams and make logic component class diagram better
prof's advise

### PR DESCRIPTION
**Pull request by wamikamalik**
Fix #189 
Fix #163 
Regarding #54
Regarding #67

### CODE REVIEW

File: docs/DG_Diagrams/LogicComponent.puml [hunk 1]
```diff
@@ -1,48 +1,57 @@
 @startuml
 
-allowmixing
+hide members
+hide circle
```
Comment (Johnson-Yee) [inline docs/DG_Diagrams/LogicComponent.puml:4]: Looks alot neater! But was just wondering what do ABC ,PQR and XYZ represent? And what about the commons component?

Comment (wamikamalik) [reply docs/DG_Diagrams/LogicComponent.puml:4]: The different command classes (ABC, XYZ, PQR) have been explained in individual notes beside the classes. As for commons component...I believe help and exit will be too trivial to mention

Comment (Johnson-Yee) [reply docs/DG_Diagrams/LogicComponent.puml:4]: Alright! This part looks good!

Comment (Johnson-Yee) [reply docs/DG_Diagrams/LogicComponent.puml:4]: Alright! This part looks good!

**Review by Johnson-Yee — APPROVED** [review]

**Comment by wamikamalik** [comment]
Regrading #149

**Pull request merged**